### PR TITLE
Add telemetry events for the sidebar

### PR DESCRIPTION
**Pull request by ianlaidlaw**
- Added telemetry events to the sidebar for sequence viewer (open and close). These are handled using a new helper class: `TelemetryHelper.js`
- When the user clicks the icon to open, it will track an open event.
- When the sidebar is closed either by clicking the icon again or clicking on the main view, it will track a close event.
- Added a new dependency `chai-spies` to help with testing.

### CODE REVIEW

File: d2l-sequence-viewer.js [hunk 1]
```diff
@@ -29,7 +30,7 @@ import { PolymerElement } from '@polymer/polymer/polymer-element.js';
 class D2LSequenceViewer extends mixinBehaviors([
 	D2L.PolymerBehaviors.Siren.EntityBehavior,
 	D2L.PolymerBehaviors.Siren.SirenActionBehaviorImpl,
-	D2L.PolymerBehaviors.SequenceViewer.LocalizeBehavior
+	D2L.PolymerBehaviors.SequenceViewer.LocalizeBehavior,
```
Comment (ianlaidlaw) [inline d2l-sequence-viewer.js]: I ran format document on the files I worked on in this pr so you may notice some unrelated formatting changes (not from me)

File: demo/index.html [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
-	<body class="d2l-typography">
-		<d2l-sequence-viewer
-			return-url="http://google.ca"
-			current-activity="data/l1-m2-activity2.json"
-			href="data/l1-m2-activity2.json"
-			token="foozleberries"
-			_module-properties='{"moduleIndex":2,"numberOfSiblingModules":6,"title":"Testo Courso","useNewProgressBar":true}'>
-		</d2l-sequence-viewer>
-	</body>
+	data-asv-css-vars="{&quot;--d2l-asv-primary-color&quot;:&quot;#c6def1&quot;,&quot;--d2l-asv-accent-color&quot;:&quot;var(--d2l-color-celestine)&quot;,&quot;--d2l-asv-text-color&quot;:&quot;var(--d2l-color-ferrite)&quot;,&quot;--d2l-asv-selected-text-color&quot;:&quot;var(--d2l-color-ferrite)&quot;,&quot;--d2l-asv-hover-color&quot;:&quot;rgba(198, 222, 241, 0.4)&quot;,&quot;--d2l-asv-border-color&quot;:&quot;rgba(23, 53, 99, 0.3)&quot;}">
```
Comment (ianlaidlaw) [inline demo/index.html]: no changes in this file, again the diff is different from formatting the files :/

File: helpers/telemetry-helper.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+		});
+
+		const eventBody = new d2lTelemetryBrowserClient.EventBody()
+			.setAction('Created')
+			.setObject(encodeURIComponent(id), 'Sequence Viewer', id);
+
+		const event = new d2lTelemetryBrowserClient.TelemetryEvent()
+			.setDate(new Date())
+			.setType('TelemetryEvent')
+			.setSourceId('learnerexperience')
```
Comment (ianlaidlaw) [inline helpers/telemetry-helper.js]: confirm the event name here with product - make sure there is no overlap

File: package.json [hunk 4]
```diff
@@ -63,6 +63,7 @@
     "bower-art-resolver": "^2.0.9",
     "browserify": "^16.2.2",
     "chai-dom": "^1.8.0",
+    "chai-spies": "^1.0.0",
```
Comment (ianlaidlaw) [inline package.json]: since this project uses chai-dom for assertions, I added this package so I can mock & spy on functions

File: test/d2l-sequence-viewer.html [hunk 5]
```diff
@@ -1,126 +1,170 @@
...
-				expect( button ).to.exist;
-				expect( button ).to.have.attribute('href', 'https://www.google.ca');
-			});
-		});
+					});
+				});
 
-		describe('basic fixture',() => {
-			let elem;
+				describe('the hamburger button', () => {
```
Comment (ianlaidlaw) [inline test/d2l-sequence-viewer.html]: formatting this file made a really big diff :/

the actual changes really start here - the tests for the side bar / hamburger button

File: d2l-sequence-viewer.js [hunk 6]
```diff
@@ -447,13 +450,20 @@ class D2LSequenceViewer extends mixinBehaviors([
 		}
 		this.$.sidebar.classList.remove('offscreen');
 
+		if (this.telemetryEndpoint) {
```
Comment (pbrown-d2l) [inline d2l-sequence-viewer.js]: Just to remove clutter, it might be worth moving this check into the telemetry-helper.js

File: helpers/telemetry-helper.js (hunk 3 above)
Comment (ChrisKraljevicD2L) [reply helpers/telemetry-helper.js]: Any update on if this `sourceId` is clear?

Comment (ianlaidlaw) [reply helpers/telemetry-helper.js]: ah yes! I got confirmation this is safe to use.
